app/models/move.rb

```diff
@@ -219,6 +220,14 @@ def vehicle_registration
     journeys.max_by(&:client_timestamp)&.vehicle&.dig('registration')
   end
 
+  def expected_time_of_arrival
+    notification_events.select { |event| event.type == 'GenericEvent::MoveNotifyPremisesOfEta' }.max_by(&:occurred_at)&.expected_at
+  end
+
+  def expected_collection_time
+    notification_events.select { |event| event.type == 'GenericEvent::MoveNotifyPremisesOfExpectedCollectionTime' }.max_by(&:occurred_at)&.expected_at
```

Comment:
Would be nice if both of these could use SQL approach to return a single event row for each case if possible.   You mentioned a potential N+1 query issue here - I'm guessing that occurs when returning multiple moves and to mitigate that you're also eager loading `notification_events` for moves? If so then that's probably at least worth mentioning in a comment for these two methods...

Reply:
Tested again and unfortunately doing this in SQL is indeed causing an n+1, I added a comment for posterity